flashinfer/norm/__init__.py

```
@@ -761,6 +761,559 @@ def fused_rmsnorm_silu(
+    major, minor = get_compute_capability(input.device)
+    sm = major * 10 + minor
+    if sm < 80:
+        raise RuntimeError("Fused DIT LayerNorm requires SM80+")
+    if (use_nvfp4 or use_mxfp8) and sm < 100:
+        raise RuntimeError(
+            f"NVFP4/MXFP8 output requires SM100+ (Blackwell), got SM{sm}"
+        )
+
+
+def _dit_ln_check_strided_tensor(
+    tensor: torch.Tensor, input: torch.Tensor, name: str
```

Ruff (0.15.11)

[error] Function argument `input` is shadowing a Python builtin

(A002)